### PR TITLE
Fahad changes

### PR DESCRIPTION
**Pull request by excellencetechnologies**

### CODE REVIEW

File: attendance/API_HR/c-hr.php [hunk 1]
```diff
@@ -2302,6 +2302,39 @@ public static function updateGooglepaySlipDriveAccessToken($google_access_token)
         return $return;
     }
 
+    // Welcome mail
+
+    public function sendWelcomeMail($mailmsg) {
+        // fetching welcome email template
+        $q = "SELECT * FROM email_templates where id=52";
```
Comment (excellencetechnologies) [inline attendance/API_HR/c-hr.php]: it should be on the basis of name, another DB can have different id.

File: attendance/API_HR/c-hr.php [hunk 2]
```diff
@@ -2302,6 +2302,39 @@ public static function updateGooglepaySlipDriveAccessToken($google_access_token)
...
+        $runQuery = self::DBrunQuery($q);
+        $row = self::DBfetchRows($runQuery);
+        $mail_body = $row[0]['body'];
```
Comment (excellencetechnologies) [inline attendance/API_HR/c-hr.php]: is $row is empty it will throw error, have some check here

File: attendance/API_HR/c-hr.php [hunk 3]
```diff
@@ -2376,7 +2409,15 @@ public static function addNewEmployee($PARAMS) { //api call
...
                     self::DBrunQuery($q1);
                     $r_error = 0;
-                    $r_message = "Employee added Successfully !!";
+                    
+                    //$r_message = "Employee added Successfully !!";
+
+                    // Added on 15-03-18 to send Welcome mail to new user
+                    if ($userID == true) {
+                        $pr_message = "Employee added Successfully !!";
+                        $tr_message = self::sendWelcomeMail($mailmsg); // call welcome mail
```
Comment (excellencetechnologies) [inline attendance/API_HR/c-hr.php]: what is $mailmsg???
u should should send userID here & in the function and you can use getUserInfo function to extract the user info in sendWelcomeMail funtion.

File: attendance/API_HR/c-hr.php [hunk 4]
```diff
@@ -2376,7 +2409,15 @@ public static function addNewEmployee($PARAMS) { //api call
...
+                        $r_message = $pr_message." ".$tr_message;
```
Comment (excellencetechnologies) [inline attendance/API_HR/c-hr.php]: why did u concat this tr_message to pr_message?

File: attendance/API_HR/c-hr.php (hunk 2 above)
Comment (excellencetechnologies) [reply attendance/API_HR/c-hr.php]: okay sir. i'll add an if statement to check for empty row

File: attendance/API_HR/c-hr.php (hunk 4 above)
Comment (excellencetechnologies) [reply attendance/API_HR/c-hr.php]: sir, to throw a pop up message that will show "employee added successfully. Email is sent"

File: attendance/API_HR/c-hr.php [hunk 5]
```diff
@@ -2302,6 +2302,142 @@ public static function updateGooglepaySlipDriveAccessToken($google_access_token)
...
+                $cc = $var['cc_detail'];
+                $bcc = $var['bcc_detail'];
+                $file_upload = $var['upload_file'];
+                
+                $mail = new PHPMailer;
+                $mail->isSMTP();
+                $mail->SMTPDebug = 0;
+                $mail->Debugoutput = 'html';
+                $mail->Host = $detail['host'];
+                $mail->Port = $detail['port'];
```
Comment (reactexcel) [inline attendance/API_HR/c-hr.php]: its post earlier??? if yes keep it as it is

File: attendance/API_HR/c-hr.php [hunk 6]
```diff
@@ -2376,7 +2512,15 @@ public static function addNewEmployee($PARAMS) { //api call
                         ( '$f_name', '$f_jobtitle', '$f_dateofjoining', $userID, '$f_dob', '$f_gender', '$f_workemail', $f_training_month ) ";
                     self::DBrunQuery($q1);
                     $r_error = 0;
-                    $r_message = "Employee added Successfully !!";
+                    
+                    //$r_message = "Employee added Successfully !!";
+
+                    // Added on 15-03-18 to send Welcome mail to new user
+                    if ($userID == true) {
```
Comment (reactexcel) [inline attendance/API_HR/c-hr.php]: what is this condition ? is user id is true i.e a boolean, what you are sending in sendWelcomeMail function?

File: attendance/API_HR/c-hr.php [hunk 7]
```diff
@@ -2376,7 +2512,15 @@ public static function addNewEmployee($PARAMS) { //api call
                         ( '$f_name', '$f_jobtitle', '$f_dateofjoining', $userID, '$f_dob', '$f_gender', '$f_workemail', $f_training_month ) ";
                     self::DBrunQuery($q1);
                     $r_error = 0;
-                    $r_message = "Employee added Successfully !!";
```
Comment (reactexcel) [inline attendance/API_HR/c-hr.php]: keep this message , as it is .

File: attendance/API_HR/c-hr.php [hunk 8]
```diff
@@ -2376,7 +2512,15 @@ public static function addNewEmployee($PARAMS) { //api call
...
+                    
+                    //$r_message = "Employee added Successfully !!";
+
+                    // Added on 15-03-18 to send Welcome mail to new user
+                    if ($userID == true) {
+                        $r_message = "Employee added Successfully !!";
```
Comment (reactexcel) [inline attendance/API_HR/c-hr.php]: append in message that 
Welcome email Sent !

File: attendance/API_HR/c-hr.php [hunk 9]
```diff
@@ -2302,6 +2302,142 @@ public static function updateGooglepaySlipDriveAccessToken($google_access_token)
...
+
+
+
+    //Welcome mail
+
+    public function sendWelcomeMail($userID) {
+        $userInfo = self::getUserInfo($userID);
+
+        // fetching welcome email template
+        $q = "SELECT * FROM email_templates where name='Welcome mail'";
```
Comment (reactexcel) [inline attendance/API_HR/c-hr.php]: change this to `New Employee Welcome Email`

File: attendance/API_HR/c-hr.php [hunk 10]
```diff
@@ -2302,6 +2302,142 @@ public static function updateGooglepaySlipDriveAccessToken($google_access_token)
...
+        }
+
+        $data = array();
+        $data['email']['subject'] = 'Welcome to Excellence Technologies';
+        $data['email']['name'] = $userInfo['name'];
+        $data['email']['body'] = $mail_body;
+        $data['email']['email_id'] = $work_email;
+        $msg = "";
+        $msg = self::sendEmail($data); 
+        // if($msg!="Message sent") {
```
Comment (reactexcel) [inline attendance/API_HR/c-hr.php]: remove commented code done by you

File: attendance/API_HR/c-hr.php [hunk 11]
```diff
@@ -2302,6 +2302,142 @@ public static function updateGooglepaySlipDriveAccessToken($google_access_token)
...
+        $r_error = 1;
+        $r_message = "";
+        $r_data = array();
+
+        $q = "SELECT * from config where type='email_detail'";
+        $r = self::DBrunQuery($q);
+        $row = self::DBfetchRow($r);
+        // convert this string into associative array
+        parse_str($row['value'], $detail);
+        include "phpmailer/PHPMailerAutoload.php";
```
Comment (reactexcel) [inline attendance/API_HR/c-hr.php]: is this Folder is added under API_HR folder, if yes why it is not in PR?

File: attendance/API_HR/c-hr.php [hunk 12]
```diff
@@ -2302,6 +2302,138 @@ public static function updateGooglepaySlipDriveAccessToken($google_access_token)
...
+
+
+
+    //New Employee Welcome Email
+
+    public function sendWelcomeMail($userID) {
+        $userInfo = self::getUserInfo($userID);
+
+        // Fetching welcome email template
+        $q = "SELECT * FROM email_templates where name='Welcome mail'";
```
Comment (reactexcel) [inline attendance/API_HR/c-hr.php]: this is not changed
change this to New Employee Welcome Email

File: attendance/API_HR/c-hr.php [hunk 13]
```diff
@@ -2302,6 +2302,138 @@ public static function updateGooglepaySlipDriveAccessToken($google_access_token)
...
+        $runQuery = self::DBrunQuery($q);
+        $row = self::DBfetchRows($runQuery);
+        if(empty($row)) {
+           echo "Template not fetched!";
+           die;
```
Comment (reactexcel) [inline attendance/API_HR/c-hr.php]: die will lead to stop the script you need to return from here and same will be reflected.

File: attendance/API_HR/c-hr.php [hunk 14]
```diff
@@ -2302,6 +2302,138 @@ public static function updateGooglepaySlipDriveAccessToken($google_access_token)
...
+        // Fetching value of Variables in Template
+        $q2 = 'Select * from template_variables';
+        $runQuery2 = self::DBrunQuery($q2);
+        $row2 = self::DBfetchRows($runQuery2);
+        foreach ($row2 as $s) {
+            $mail_body = str_replace($s['name'], $s['value'], $mail_body);
+        }
+
+        $data = array();
+        $data['email']['subject'] = 'Welcome to Excellence Technologies';
```
Comment (reactexcel) [inline attendance/API_HR/c-hr.php]: can u see and check , i think subject also comes from template? if yes the no need to hard code here.

File: attendance/API_HR/c-hr.php [hunk 15]
```diff
@@ -2302,6 +2302,140 @@ public static function updateGooglepaySlipDriveAccessToken($google_access_token)
...
+
+    public function sendWelcomeMail($userID) {
+        $userInfo = self::getUserInfo($userID);
+
+        // Fetching New Employee Welcome Email template
+        $q = "SELECT * FROM email_templates where name='New Employee Welcome Email'";
+        $runQuery = self::DBrunQuery($q);
+        $row = self::DBfetchRows($runQuery);
+        if(empty($row)) {
+           $r_message = "Template not fetched!";
```
Comment (reactexcel) [inline attendance/API_HR/c-hr.php]: Change - Template not fetched
to 
Warning - New Employee Welcome Email template not found!!

**Pull request merged**